src/main/java/org/jabref/preferences/SearchPreferences.java

```diff
@@ -7,11 +7,13 @@
     private final SearchDisplayMode searchDisplayMode;
     private final boolean isCaseSensitive;
     private final boolean isRegularExpression;
+    private final boolean isFulltext;
 
-    public SearchPreferences(SearchDisplayMode searchDisplayMode, boolean isCaseSensitive, boolean isRegularExpression) {
+    public SearchPreferences(SearchDisplayMode searchDisplayMode, boolean isCaseSensitive, boolean isRegularExpression, boolean isFulltext) {
```

Comment:
enums instead of bools could be used here too

Reply:
I kept it here because I don't want to mess with peoples Search-Preferences (stored). How do you usually handle such updates?

Reply:
Have a look at the PreferencesMigration